pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/when/impl/BuildingChangeConditional.java

```diff
@@ -0,0 +1,347 @@
+     * @return authorDisplayName
+     * @see ContributorMetadataAction#getContributor()
+     */
+    public String getAuthorX() {
+        return authorX;
+    }
+
+    @DataBoundSetter
+    public void setAuthorX(String authorX) {
+        this.authorX = authorX;
+    }
+
+    /**
+     * CHANGE_AUTHOR_DISPLAY_NAME regular expression matching.
+     * @return authorDisplayName
+     * @see ContributorMetadataAction#getContributorDisplayName()
+     */
+    public String getAuthorDisplayNameX() {
+        return authorDisplayNameX;
+    }
+
+    @DataBoundSetter
+    public void setAuthorDisplayNameX(String authorDisplayNameX) {
+        this.authorDisplayNameX = authorDisplayNameX;
+    }
+
+    /**
+     * CHANGE_AUTHOR_EMAIL ant style matching.
+     * @return authorEmail
+     * @see ContributorMetadataAction#getContributorEmail()
+     */
+    public String getAuthorEmail() {
+        return authorEmail;
+    }
+
+    @DataBoundSetter
+    public void setAuthorEmail(String authorEmail) {
+        this.authorEmail = authorEmail;
+    }
+
+    /**
+     * CHANGE_AUTHOR_EMAIL regular expression matching.
+     * @return authorEmailX
+     * @see ContributorMetadataAction#getContributorEmail()
+     */
+    public String getAuthorEmailX() {
+        return authorEmailX;
+    }
+
+    @DataBoundSetter
+    public void setAuthorEmailX(String authorEmailX) {
+        this.authorEmailX = authorEmailX;
+    }
+
+    public boolean matches(EnvVars vars) {
+        if (!CHANGE_ID.exists(vars)) {
+            return false; //Not a change
+        }
+        try {
+            check(id, Comparator.ANT, CHANGE_ID, vars);
+            check(target, Comparator.ANT, CHANGE_TARGET, vars);
+            check(branch, Comparator.ANT, CHANGE_BRANCH, vars);
+            check(fork, Comparator.ANT, CHANGE_FORK, vars);
+            check(urlX, Comparator.REG, CHANGE_URL, vars);
+            check(title, Comparator.ANT, CHANGE_TITLE, vars);
+            check(titleX, Comparator.REG, CHANGE_TITLE, vars);
+            check(author, Comparator.ANT, CHANGE_AUTHOR, vars);
+            check(authorX, Comparator.REG, CHANGE_AUTHOR, vars);
+            check(authorDisplayName, Comparator.ANT, CHANGE_AUTHOR_DISPLAY_NAME, vars);
+            check(authorDisplayNameX, Comparator.REG, CHANGE_AUTHOR_DISPLAY_NAME, vars);
+            check(authorEmail, Comparator.ANT, CHANGE_AUTHOR_EMAIL, vars);
+            check(authorEmailX, Comparator.REG, CHANGE_AUTHOR_EMAIL, vars);
+            return true;
+        } catch (AssertionError e) {
+            return false;
+        }
+    }
+
+    private void check(String pattern, Comparator comparator, Names varName, EnvVars vars) {
+        if (isNotEmpty(pattern)) {
+            if (!comparator.compare(pattern, varName.get(vars))) {
+                throw new AssertionError(varName.name() + "("+varName.get(vars)+")" + " does not match " + pattern);
+            }
+        }
+    }
+
+    @Extension
+    @Symbol("buildingChange")
```

Comment:
I don't love the symbol, but can't think of a better one yet.

Reply:
I needed something that was a bit further away from `changeset` and `changelog` than just `change`.

Reply:
I definitely understand - I'm just saying we need more to explain what this means than just `(something)Change` or `change`. After all, any SCM-triggered build on `master` has a *change*, just not a *change request*.

Reply:
I was thinking of `cr` at first but feared it was too short. And then when we decided on `buildingTag` I though I would follow that naming scheme :) But I agree, it's not very beautiful.

Reply:
Yeah, this is gonna need more thought and discussion. @michaelneale, @bitwiseman - thoughts?

Reply:
is changeRequest the generalisation of pull request for all SCMs?

Reply:
Yes, branch-api and scm-api generalize all of them as change request. Hence the environment variable names.